ParkBlazerApp/src/pages/Rent.tsx

```diff
@@ -0,0 +1,199 @@
+
+    startDate = adjustMinutes(startDate);
+    endDate = adjustMinutes(endDate);
+    setStartDate(startDate);
+    setEndDate(endDate);
+
+    async function getParkingspot() {
+      await initParkingSpaces();
+      const parkingspotDetails = parkingspaces.find(ps => ps.parkingspot_id === Number(id));
+      if (parkingspotDetails) {
+        setParkingspot(parkingspotDetails);
+        setRestrictedDates(await getReservedDates(parkingspotDetails.private_parkingspot_id!));
+      }
+    }
+
+    async function fetchReviews() {
+      try {
+        const response = await axios.post('https://server-y2mz.onrender.com/api/get_reviews_of_parkingspot', {
+          parkingspot_id: id
+        });
+        const reviewDetails = response.data;
+        setReviews(reviewDetails);
+      } catch (error: any) {
+        console.error(error);
+      }
+    };
+
+    fetchReviews();
+    getParkingspot();
+  }, []);
+
+  useEffect(() => {
+    if (!start_date && !end_date) return;
+    if (!validateDates(start_date!, end_date!)) {
+      setStartDate(null);
+      setEndDate(null);
+    } else {
+      setRentTimeInHours(parseFloat(((end_date!.getTime() - start_date!.getTime()) / (1000 * 60 * 60)).toFixed(2)));
+    }
+  }, [restrictedDates]);
+
+  function calculateAverageRating(reviews: any[]) {
+    if (!reviews || reviews.length === 0) return "No reviews yet";
+
+    const totalRating = reviews.reduce((acc, curr) => acc + curr.rating, 0);
+    const averageRating = Math.round((totalRating / reviews.length) * 2) / 2;
+
+    return (
+      <StarRating rating={averageRating} />
+    );
+  };
+
+  function validateDates(startDate: Date, endDate: Date) {
+    if (isNaN(startDate.getTime()) || isNaN(endDate.getTime())) {
+      return false;
+    }
+    else if (findRestrictedDateInRange(startDate, endDate, restrictedDates!)) {
+      return false;
+    }
+    else if (adjustDateToAvailability(startDate, parkingspace!, true).toDateString() !== startDate.toDateString() || adjustDateToAvailability(endDate, parkingspace!, false).toDateString() !== endDate.toDateString()) {
+      return false;
+    }
+    return true;
+  }
+
+  function handleRedirect(location: string) {
+    history.push(location);
+    window.location.reload();
+  }
+
+  async function handleRentClick(e: any) {
+    try {
+      await axios.post('https://server-y2mz.onrender.com/api/create_reservation', {
+        start_date: start_date?.toISOString(),
+        end_date: end_date?.toISOString(),
+        private_parkingspot_id: parkingspace?.private_parkingspot_id,
+        amount: Number((parkingspace?.price_per_hour! * rentTimeInHours * 1.29).toFixed(2)),
+        payment_method: paymentMethod,
+        userId: AuthService.getToken()
+      }, {
+        headers: {
+          Authorization: `Bearer ${AuthService.getToken()}`
+        }
+      });
+    } catch (error: any) {
+      console.error(error);
+    }
+    handleRedirect("/home");
+  }
```

Comment:
Ich würde die Reservierungen in eine eigene Datei tun denn in parkingspaces.ts wurde auch schon ein fetch für die Reservierungen eingefügt, dann sind die Reservierungen Zentraler an einem Ort